src/CodeGen_LLVM.cpp

```diff
@@ -2956,7 +2944,26 @@ void CodeGen_LLVM::visit(const Call *op) {
                 }
             }
         }
-
+    } else if (op->is_intrinsic(Call::saturating_add) || op->is_intrinsic(Call::saturating_sub)) {
+        internal_assert(op->args.size() == 2);
+        std::string intrin;
+        if (op->type.is_int()) {
+            intrin = "llvm.s";
+        } else {
+            internal_assert(op->type.is_uint());
+            intrin = "llvm.u";
+        }
+        if (op->is_intrinsic(Call::saturating_add)) {
+            intrin += "add.sat.";
+        } else {
+            internal_assert(op->is_intrinsic(Call::saturating_sub));
+            intrin += "sub.sat.";
+        }
+        if (op->type.lanes() > 1) {
+            intrin += "v" + std::to_string(op->type.lanes());
+        }
+        intrin += "i" + std::to_string(op->type.bits());
+        value = call_intrin(op->type, op->type.lanes(), intrin, op->args);
```

Comment:
The lanes arg is supposed to be the native lanes of the intrinsic, so vector width legalization must have already happened somewhere?

Reply:
LLVM's saturating add/sub takes any number of lanes, it must do the legalization internally, just like a regular add/sub/etc. I think we are supposed to treat these like any other arithmetic operator in LLVM.

Reply:
Ah, I didn't realize. Weird that it's stringly typed.

Reply:
Going to resolve the related comments, please reopen if needed.